pdns/recursordist/test-recpacketcache_cc.cc

```diff
@@ -80,6 +82,66 @@ BOOST_AUTO_TEST_CASE(test_recPacketCacheSimple)
   BOOST_CHECK_EQUAL(rpc.size(), 0U);
 }
 
+BOOST_AUTO_TEST_CASE(test_recPacketCacheSimpleWithRefresh)
+{
+  RecursorPacketCache::s_refresh_ttlperc = 70;
+  RecursorPacketCache rpc(1000);
+  string fpacket;
+  unsigned int tag = 0;
+  uint32_t age = 0;
+  uint32_t qhash = 0;
+  uint32_t ttd = 3600;
+  BOOST_CHECK_EQUAL(rpc.size(), 0U);
+
+  DNSName qname("www.powerdns.com");
+  vector<uint8_t> packet;
+  DNSPacketWriter pw(packet, qname, QType::A);
+  pw.getHeader()->rd = true;
+  pw.getHeader()->qr = false;
+  pw.getHeader()->id = dns_random_uint16();
+  string qpacket((const char*)&packet[0], packet.size());
+  pw.startRecord(qname, QType::A, ttd);
+
+  time_t now = time(nullptr);
+
+  BOOST_CHECK_EQUAL(rpc.getResponsePacket(tag, qpacket, qname, QType::A, QClass::IN, now, &fpacket, &age, &qhash), false);
+
+  ARecordContent ar("127.0.0.1");
+  ar.toPacket(pw);
+  pw.commit();
+  string rpacket((const char*)&packet[0], packet.size());
+
+  rpc.insertResponsePacket(tag, qhash, string(qpacket), qname, QType::A, QClass::IN, string(rpacket), now, ttd, vState::Indeterminate, boost::none, false);
+  BOOST_CHECK_EQUAL(rpc.size(), 1U);
+  uint32_t qhash2 = 0;
+  bool found = rpc.getResponsePacket(tag, qpacket, qname, QType::A, QClass::IN, now, &fpacket, &age, &qhash2);
+  BOOST_CHECK_EQUAL(found, true);
+  BOOST_CHECK_EQUAL(qhash, qhash2);
+  BOOST_CHECK_EQUAL(fpacket, rpacket);
+
+  BOOST_REQUIRE_EQUAL(getTaskSize(), 0U);
+
+  // Half of time has passed, no task should have been submitted
```

Comment:
I'm not sure I agree with this:
- `RecursorPacketCache::s_refresh_ttlperc` is set to `70` which, looking at the code, is equivalent to setting `refresh-on-ttl-perc` to `70`
- the documentation states `Whenever a record is fetched from the packet or record cache and only refresh-on-ttl-perc percent or less of its original TTL is left [...]`
- at this point, only 50% of the original TTL is left, which seems less than 70% to me?

Reply:
Oops, I might have confused myself.... will revisit.

Reply:
Unconfused myself. The original deadline computation is OK. Removed that commit and fixed the test, force pushed to now have the wrong commit in history.